specs/agents/configuration.md

```diff
@@ -82,6 +82,8 @@ Agents should poll the APM Server for config periodically by sending an HTTP req
 
 The server will respond with a JSON object, where each key maps a config attribute to a string value. The string value should be interpreted the same as if it were passed in via an environment variable. Upon receiving these config changes, the agent will update its configuration dynamically, overriding any config previously specified. That is, config via Kibana takes highest precedence.
 
+To ensure graceful recovery if the APM Server is offline or there is a network partition, agents should only retry central configuration requests occasionally, and should add a modest random delay when retrying to avoid thundering herd issues when the partition resolves.
```

Comment:
Isn't this already covered by this part of the spec?

> For older servers, or for whatever reason a 7.3+ server does not respond with that header (or it is invalid), agents should retry after 5 minutes.

Or is the issue that the server/proxy may return a header like `Cache-Control: max-age=0`? Should we treat 0 the same as a missing cache-control header? Should we we have a static minimum value?

In contrast to the intake api, I don't feel like we necessarily need an exponential backoff as the default polling interval to central config is 30s. So it's hard for the server to be overwhelmed by that rate. The max backoff for the intake API is 36s.

Reply:
Yes, that makes sense. The proxy returns `Cache-Control: max-age=0` in the error case presented by the customer. I think a static minimum value makes some sense, but what that is I don't have an opinion on. Anything >0 should be fine I think.

I think you're right about the backoff, probably not a big issue. Will update text with above changes 👍

Reply:
Looks good now. But please move it to the `Dealing with errors` section that already describes how to deal with different and erroneous Cache-Control headers.

Reply:
Thanks, updated!